crates/sui-graphql-client/src/lib.rs

```diff
@@ -965,239 +964,205 @@ mod tests {
+        assert!(chain_id.is_ok());
     }
 
     #[tokio::test]
     async fn test_reference_gas_price_query() {
-        for (n, _) in NETWORKS.iter() {
-            let client = Client::new(n).unwrap();
-            let rgp = client.reference_gas_price(None).await;
-            assert!(
-                rgp.is_ok(),
-                "Reference gas price query failed for network: {n}"
-            );
-        }
+        let client = Client::new_localhost();
+        let rgp = client.reference_gas_price(None).await;
+        assert!(
+            rgp.is_ok(),
+            "Reference gas price query failed for local network"
+        );
     }
 
     #[tokio::test]
     async fn test_protocol_config_query() {
-        for (n, _) in NETWORKS {
-            let client = Client::new(n).unwrap();
-            let pc = client.protocol_config(None).await;
-            assert!(pc.is_ok());
-
-            // test specific version
-            let pc = client.protocol_config(Some(50)).await;
-            assert!(pc.is_ok());
-            let pc = pc.unwrap();
-            if let Some(pc) = pc {
-                assert_eq!(
-                    pc.protocol_version, 50,
-                    "Protocol version query mismatch for network: {n}. Expected: 50, received: {}",
-                    pc.protocol_version
-                );
-            }
+        let client = Client::new_localhost();
+        let pc = client.protocol_config(None).await;
+        assert!(pc.is_ok());
+
+        // test specific version
+        let pc = client.protocol_config(Some(50)).await;
+        assert!(pc.is_ok());
+        let pc = pc.unwrap();
+        if let Some(pc) = pc {
+            assert_eq!(
+                pc.protocol_version, 50,
+                "Protocol version query mismatch for local network. Expected: 50, received: {}",
+                pc.protocol_version
+            );
         }
     }
 
     #[tokio::test]
     async fn test_service_config_query() {
-        for (n, _) in NETWORKS {
-            let client = Client::new(n).unwrap();
-            let sc = client.service_config().await;
-            assert!(sc.is_ok(), "Service config query failed for network: {n}");
-        }
+        let client = Client::new_localhost();
+        let sc = client.service_config().await;
+        assert!(sc.is_ok(), "Service config query failed for local network");
     }
 
     #[tokio::test]
     async fn test_active_validators() {
-        for (n, _) in NETWORKS {
-            let client = Client::new(n).unwrap();
-            let av = client.active_validators(None, None).await;
-            assert!(
-                av.is_ok(),
-                "Active validators query failed for network: {n}. Error: {}",
-                av.unwrap_err()
-            );
+        // this test requires to have at least epoch that finished, so wait for 10s just to be on
+        // the safe side
+        tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
```

Comment:
Can we do this sleep outside of the test and instead on the startup of the local network?

Reply:
Oh, yes, that sounds like a better idea!